inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java

```diff
@@ -0,0 +1,660 @@
+                        String message = t.getMessage();
+                        if (message != null && (message.contains("No available queue for topic")
+                                || message.contains("The brokers of topic are all forbidden"))) {
+                            illegalTopicMap.put(topic, System.currentTimeMillis() + 60 * 1000);
+                            logger.info("IllegalTopicMap.put " + topic);
+                            continue;
+                        } else {
+                            try {
+                                Thread.sleep(100);
+                            } catch (InterruptedException e) {
+                                //ignore..
+                            }
+                        }
+                    }
+                    resendEvent(es, decrementFlag);
+                }
+            }
+        }
+
+        private boolean sendMessage(Event event, String topic, EventStat es)
+                throws TubeClientException, InterruptedException {
+            MessageProducer producer = getProducer(topic);
+            if (producer == null) {
+                illegalTopicMap.put(topic, System.currentTimeMillis() + 30 * 1000);
+                logger.error("Get producer is null, topic:{}", topic);
+                return false;
+            }
+
+            Message message = new Message(topic, event.getBody());
+            message.setAttrKeyVal("auditIp", localIp);
+            String streamId = "";
+            String groupId = "";
+            if (event.getHeaders().containsKey(org.apache.inlong.audit.consts.AttributeConstants.INLONG_STREAM_ID)) {
+                streamId = event.getHeaders().get(org.apache.inlong.audit.consts.AttributeConstants.INLONG_STREAM_ID);
+                message.setAttrKeyVal(org.apache.inlong.audit.consts.AttributeConstants.INLONG_STREAM_ID, streamId);
+            }
+            if (event.getHeaders().containsKey(org.apache.inlong.audit.consts.AttributeConstants.INLONG_GROUP_ID)) {
+                groupId = event.getHeaders().get(org.apache.inlong.audit.consts.AttributeConstants.INLONG_GROUP_ID);
+                message.setAttrKeyVal(org.apache.inlong.audit.consts.AttributeConstants.INLONG_GROUP_ID, groupId);
+            }
+
+            logger.debug("producer start to send msg...");
+            producer.sendMessage(message, new MyCallback(es));
+
+            illegalTopicMap.remove(topic);
+            return true;
+        }
+    }
+
+    public class MyCallback implements MessageSentCallback {
+
+        private org.apache.inlong.audit.sink.EventStat myEventStat;
+        private long sendTime;
+
+        public MyCallback(org.apache.inlong.audit.sink.EventStat eventStat) {
+            this.myEventStat = eventStat;
+            this.sendTime = System.currentTimeMillis();
+        }
+
+        @Override
+        public void onMessageSent(final MessageSentResult result) {
+            if (result.isSuccess()) {
```

Comment:
It is recommended to use guard statements to reduce the cyclomatic complexity of the method.

Reply:
done